polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -1047,21 +1047,21 @@ static NamedAttrList getSYCLMethodOpAttrs(OpBuilder &Builder,
 ///
 /// This function relies on how arguments are casted to perform a function call.
 /// Should be updated if this changes.
-static llvm::Optional<mlir::sycl::SYCLCastOp> trackSYCLCast(Value Val) {
+static Operation *trackCasts(Value Val) {
   auto *const DefiningOp = Val.getDefiningOp();
```

Comment:
auto -> Operation

Reply:
I'd say the type is not needed here, as `getDefiningOp` is quite straightforward.

Reply:
That is Ok. I was curious and so I checked the MLIR codebase. Grepping for "=.*getDefiningOp\(\);" in the MLIR codebase shows that the *vast* majority of instances declare the LHS using `Operation*`. Just a data point.